notification-service/src/main/java/org/hypertrace/alert/engine/notification/service/NotificationEventProcessorService.java

```diff
@@ -48,15 +46,10 @@ protected void doInit() {}
   @Override
   protected void doStart() {
     while (true) {
-      try {
-        ConsumerRecords<String, ByteBuffer> records =
-            consumer.poll(Duration.ofMillis(CONSUMER_POLL_TIMEOUT_MS));
-        for (ConsumerRecord<String, ByteBuffer> record : records) {
-          notificationEventProcessor.process(
-              MetricAnomalyNotificationEvent.fromByteBuffer(record.value()));
-        }
-      } catch (IOException e) {
-        LOGGER.error("Exception processing record", e);
+      ConsumerRecords<String, MetricAnomalyNotificationEvent> records =
```

Comment:
How would the serde issue be handled? In such cases, service will die, right?

Reply:
What do you mean by Serde issue?